src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -189,10 +192,15 @@ protected void doInit() {
 
             final Map<String, List<MapEntry>> newResolveMapsMap = new ConcurrentHashMap<>();
 
+            isAliasMapInitialized = false;
             //optimization made in SLING-2521
             if (this.factory.isOptimizeAliasResolutionEnabled()) {
-                final Map<String, Map<String, String>> aliasMap = this.loadAliases(resolver);
-                this.aliasMap = aliasMap;
+    			new Thread(new Runnable(){
```

Comment:
What happens when the component is stopped/started quickly in succession? How do you ensure that no leftover threads are left running?

Reply:
@rombert  - the thread at the moment would afair simply complete - certainly quickly starting and stopping would let threads pile up - so I'll capture the thread reference and interrupt in stop() (everything else seems to be overly complicated for a rather virtual case).